Synchronize fetch WPT

rreno opened the pull request:
#### 30e82dd4126d70741ec62f55bd9cc5d2f4ab4b4c
<pre>
Synchronize fetch WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=248412">https://bugs.webkit.org/show_bug.cgi?id=248412</a>
rdar://102725822

Reviewed by Tim Nguyen.

Synchronize fetch WPT as of upstream revision 22ecfc9bac05cadf8ed5e47a591f1d7e74fdc2c2

* LayoutTests/platform/glib/imported/w3c/web-platform-tests/fetch/orb/tentative/img-mime-types-coverage.tentative.sub-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/glib/TestExpectations:

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/serviceworker-intercepted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/serviceworker-intercepted.https.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.js: Added.
(async getAuthorizationHeaderValue):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.serviceworker.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.sharedworker.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.worker.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/resources/dump-authorization-header.py: Added.
(main):
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.js:
(promise_test.async let):
(promise_test):
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.serviceworker.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.sharedworker.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.worker.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/js-unlabeled.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/png-mislabeled-as-html.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/png-mislabeled-as-html.png.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/png-unlabeled.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/w3c-import.log.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-mime-types-coverage.tentative.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-mime-types-coverage.tentative.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-mislabeled-as-html.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-mislabeled-as-html.sub-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-mislabeled-as-html.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-unlabeled.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-unlabeled.sub-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-unlabeled.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.https.window.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.https.window.js: Added.
(setup):
(makePreflightTests):
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.window.js:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/resources/executor.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/resources/support.sub.js:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/w3c-import.log:

Canonical link: <a href="https://commits.webkit.org/257191@main">https://commits.webkit.org/257191@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb3667035b9ad82aacabaffd3fc786e45211e699

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98164 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7377 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31309 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107624 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167886 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7871 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/36142 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/90753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/104213 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103816 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84760 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/90753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/87790 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/89489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/90753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1341 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/1300 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/22424 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4951 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6174 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2635 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41846 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Please undo changes to LayoutTests/imported/w3c/resources/import-expectations.json and wait for green EWS

rreno committed:
Synchronize fetch WPT
https://bugs.webkit.org/show_bug.cgi?id=248412
rdar://102725822

Reviewed by Tim Nguyen.

Synchronize fetch WPT as of upstream revision 22ecfc9bac05cadf8ed5e47a591f1d7e74fdc2c2

* LayoutTests/platform/glib/imported/w3c/web-platform-tests/fetch/orb/tentative/img-mime-types-coverage.tentative.sub-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/glib/TestExpectations:

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/serviceworker-intercepted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/serviceworker-intercepted.https.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.js: Added.
(async getAuthorizationHeaderValue):
(promise_test.async test):
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.serviceworker.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.sharedworker.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/authentication-redirection.any.worker.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/resources/dump-authorization-header.py: Added.
(main):
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.js:
(promise_test.async let):
(promise_test):
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-static-error.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.serviceworker.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.sharedworker.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.worker.html:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/js-unlabeled.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/png-mislabeled-as-html.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/png-mislabeled-as-html.png.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/png-unlabeled.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/resources/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/api/credentials/w3c-import.log.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-mime-types-coverage.tentative.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-mime-types-coverage.tentative.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-mislabeled-as-html.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-mislabeled-as-html.sub-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-mislabeled-as-html.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-unlabeled.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-unlabeled.sub-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/img-png-unlabeled.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.https.window.html: Copied from LayoutTests/imported/w3c/web-platform-tests/fetch/content-encoding/gzip-body.any.html.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.https.window.js: Added.
(setup):
(makePreflightTests):
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.window.js:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/resources/executor.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/resources/support.sub.js:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/w3c-import.log:

Canonical link: https://commits.webkit.org/257191@main
